src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -0,0 +1,310 @@
+        return f"{type(self).__name__}(value={self._base_value}, exponent=0)"
+
+    def __str__(self) -> str:
+        """Return a string representation of this quantity.
+
+        Returns:
+            A string representation of this quantity.
+        """
+        return self.__format__("")
+
+    def __format__(self, __format_spec: str) -> str:
+        """Return a formatted string representation of this quantity.
+
+        If specified, must be of this form: `[0].{precision}`.  If a 0 is not given, the
+        trailing zeros will be omitted.  If no precision is given, the default is 3.
+
+        Example:
+            ```python
+            from frequenz.sdk.timeseries._quantities import Current
+            c = Current.from_amperes(1.2345)
+            assert f"{c:.2}" == "1.23 A"
+            c = Current.from_milliamperes(1.2345)
+            assert f"{c:.6}" == "1.2345 mA"
+            ```
+
+        Args:
+            __format_spec: The format specifier.
+
+        Returns:
+            A string representation of this quantity.
+
+        Raises:
+            ValueError: If the given format specifier is invalid.
+        """
+        keep_trailing_zeros = False
+        if __format_spec != "":
+            fspec_parts = __format_spec.split(".")
+            if (
+                len(fspec_parts) != 2
+                or fspec_parts[0] not in ("", "0")
+                or not fspec_parts[1].isdigit()
+            ):
+                raise ValueError(
+                    "Invalid format specifier. Must be empty or `[0].{precision}`"
+                )
+            if fspec_parts[0] == "0":
+                keep_trailing_zeros = True
+            precision = int(fspec_parts[1])
+        else:
+            precision = 3
+        abs_value = abs(self._base_value)
+        exponent = math.floor(math.log10(abs_value))
+        unit_place = exponent - exponent % 3
+        if unit_place < min(self._exponent_unit_map):
+            unit = self._exponent_unit_map[min(self._exponent_unit_map.keys())]
+            unit_place = min(self._exponent_unit_map)
+        elif unit_place > max(self._exponent_unit_map):
+            unit = self._exponent_unit_map[max(self._exponent_unit_map.keys())]
+            unit_place = max(self._exponent_unit_map)
+        else:
+            unit = self._exponent_unit_map[unit_place]
+        value_str = f"{self._base_value / 10 ** unit_place:.{precision}f}"
+        stripped = value_str.rstrip("0").rstrip(".")
+        if not keep_trailing_zeros:
+            value_str = stripped
+        unit_str = unit if stripped != "0" else self._exponent_unit_map[0]
+        return f"{value_str} {unit_str}"
+
+    def __gt__(self, other: Self) -> bool:
+        """Return whether this quantity is greater than another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is greater than another.
+        """
+        return self._base_value > other._base_value
```

Comment:
After reviewing the whole PR, I'm pretty sure we need to add runtime checks for this, because also then the type is just a general `Quantity` `mypy` won't be able to do checks.